e2e/express.e2e-spec.ts

```diff
@@ -0,0 +1,70 @@
+import { NestFactory } from '@nestjs/core';
+import * as SwaggerParser from 'swagger-parser';
+import { DocumentBuilder, SwaggerModule } from '../lib';
+import { ApplicationModule } from './src/app.module';
+import {
+  ExpressAdapter,
+  NestExpressApplication
+} from '@nestjs/platform-express';
+
+describe('Express Swagger', () => {
+  let app: NestExpressApplication;
+  let builder: DocumentBuilder;
+
+  beforeEach(async () => {
+    app = await NestFactory.create<NestExpressApplication>(
+      ApplicationModule,
+      new ExpressAdapter(),
+      { logger: false }
+    );
+
+    builder = new DocumentBuilder()
+      .setTitle('Cats example')
+      .setDescription('The cats API description')
+      .setVersion('1.0')
+      .addTag('cats')
+      .addBasicAuth()
+      .addBearerAuth()
+      .addOAuth2()
+      .addApiKey()
+      .addApiKey({ type: 'apiKey' }, 'key1')
+      .addApiKey({ type: 'apiKey' }, 'key2')
+      .addCookieAuth()
+      .addSecurityRequirements('bearer')
+      .addSecurityRequirements({ basic: [], cookie: [] });
+  });
+
+  it('should produce a valid OpenAPI 3.0 schema', async () => {
+    const document = SwaggerModule.createDocument(app, builder.build());
+    const doc = JSON.stringify(document, null, 2);
+
+    try {
+      let api = await SwaggerParser.validate(document as any);
+      console.log(
+        'API name: %s, Version: %s',
+        api.info.title,
+        api.info.version
+      );
+      expect(api.info.title).toEqual('Cats example');
+    } catch (err) {
+      console.log(doc);
+      expect(err).toBeUndefined();
```

Comment:
I Suppose this was a trick to have the `doc` logged only in the cases where the test fails? It took me a while to get this (if I understood this correctly), so I would suggest to make this test a bit easier to understand by removing the try..catch block and always logging the doc. Something like:
```ts
 const document = SwaggerModule.createDocument(app, builder.build());
 const doc = JSON.stringify(document, null, 2);
 
 console.log(doc)
 
 const api = await SwaggetParser.validate(document)
 console.log('API name: ...')
 expect(api.info.title).toEqual('Cats example');
```  

Btw, I would also suggest to have the 'Cats example' string inside a constant to avoid matching against magical strings. you could replace it by `TEST_API_TITLE`

Reply:
it's almost a blind copy of existing fastify.e2e-spec.ts =)

Reply:
Hmmm I see, but if the `validate` function throws an error the test would fail as expected, right?